addons/website_slides/controllers/slide.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import logging
+import werkzeug
+
+from odoo import http, _
+from odoo.exceptions import AccessError, UserError
+from odoo.http import request
+from odoo.addons.http_routing.models.ir_http import slug
+from odoo.addons.website.models.ir_http import sitemap_qs2dom
+
+
+_logger = logging.getLogger(__name__)
+
+class SlideController(http.Controller):
+  
+
+    @http.route('/get/quiz', type="json", auth="public", website=True)
+    def get_quiz(self, **kw):
+        if 'slide_id' in kw:
+            slide = request.env['slide.slide'].browse(kw['slide_id'])
+            slide_user = request.env['slide.slide.partner'].search([('slide_id', '=', slide.id), ('partner_id', '=', request.env.user.partner_id.id)])
+            possible_points = ["quiz_first_attempt_reward","quiz_second_attempt_reward","quiz_third_attempt_reward"]
+            res = {
+                'questions':[{'title': question.question,'id': question.id,'answers': [{'text': answer.text_value, 'correct':answer.is_correct,'id': answer.id} for answer in question.answer_ids]} for question in slide.question_ids ],
+                'nb_attempts': slide_user.quiz_attempts_count if slide_user else 0,
+                'possible_rewards': [slide.quiz_first_attempt_reward, slide.quiz_second_attempt_reward, slide.quiz_third_attempt_reward, slide.quiz_fourth_attempt_reward],
+                'reward': (slide[possible_points[slide_user.quiz_attempts_count]] if slide_user.quiz_attempts_count < 3 else slide.quiz_fourth_attempt_reward) if slide_user else slide.quiz_first_attempt_reward
```

Comment:
This makes my head hurt a little bit. Maybe write a comment for this or refactor?

Reply:
Refactored a bit